error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/JUnitFactoryMethodDeclaration.java

```diff
@@ -0,0 +1,285 @@
+    state.reportMatch(
+        buildDescription(tree)
+            .setMessage(
+                String.format(
+                    "The test cases should be supplied by a method named `%s` (but note that %s)",
+                    suggestedName, reason))
+            .build());
+  }
+
+  private ImmutableList<Description> getReturnStatementCommentFixes(
+      MethodTree testMethod, VisitorState state, MethodTree factoryMethod) {
+    ImmutableList<String> parameterNames =
+        testMethod.getParameters().stream()
+            .map(VariableTree::getName)
+            .map(Object::toString)
+            .collect(toImmutableList());
+
+    String expectedComment = parameterNames.stream().collect(joining(", ", "/* { ", " } */"));
+
+    List<? extends StatementTree> statements = factoryMethod.getBody().getStatements();
+
+    Stream<? extends StatementTree> returnStatementsNeedingComment =
+        Streams.mapWithIndex(statements.stream(), IndexedStatement::new)
+            .filter(
+                indexedStatement -> indexedStatement.getStatement().getKind() == Tree.Kind.RETURN)
+            .filter(
+                indexedStatement ->
+                    !hasExpectedComment(
+                        factoryMethod,
+                        expectedComment,
+                        statements,
+                        indexedStatement.getIndex(),
+                        state))
+            .map(IndexedStatement::getStatement);
+
+    return returnStatementsNeedingComment
+        .map(
+            s ->
+                buildDescription(s)
+                    .setMessage(
+                        "The return statement should be prefixed by a comment giving the names of the test case parameters")
+                    .addFix(SuggestedFix.prefixWith(s, expectedComment + "\n"))
+                    .build())
+        .collect(toImmutableList());
+  }
+
+  private static boolean hasExpectedComment(
+      MethodTree factoryMethod,
+      String expectedComment,
+      List<? extends StatementTree> statements,
+      long statementIndex,
+      VisitorState state) {
+    int startPosition =
+        statementIndex > 0
+            ? state.getEndPosition(statements.get((int) statementIndex - 1))
+            : ASTHelpers.getStartPosition(factoryMethod);
+    int endPosition = state.getEndPosition(statements.get((int) statementIndex));
+
+    ImmutableList<Tokens.Comment> comments =
+        extractReturnStatementComments(state, startPosition, endPosition);
+
+    return comments.stream()
+        .map(Tokens.Comment::getText)
+        .anyMatch(comment -> comment.equals(expectedComment));
+  }
+
+  private static ImmutableList<Tokens.Comment> extractReturnStatementComments(
+      VisitorState state, int startPosition, int endPosition) {
+    return state.getOffsetTokens(startPosition, endPosition).stream()
+        .filter(t -> t.kind() == Tokens.TokenKind.RETURN)
```

Comment:
```suggestion
        .filter(t -> t.kind() == TokenKind.RETURN)
```
We can remove some of the qualifiers :). Also `Tokens.Comment`.